src/Symfony/Bundle/FrameworkBundle/Routing/RouteCollectionBuilder.php

```diff
@@ -0,0 +1,465 @@
+
+    private $controllerClass;
+
+    /**
+     * @param LoaderInterface $loader
+     */
+    public function __construct(LoaderInterface $loader)
+    {
+        $this->loader = $loader;
+    }
+
+    /**
+     * Import an external routing resource, like a file.
+     *
+     * Returns a RouteCollectionBuilder so you can continue to tweak options on the routes.
+     *
+     * @param mixed  $resource
+     * @param string $prefix
+     * @param string $type
+     *
+     * @return RouteCollectionBuilder
+     */
+    public function import($resource, $prefix = null, $type = null)
+    {
+        /** @var RouteCollection $subCollection */
+        $subCollection = $this->resolve($resource, $type)->load($resource, $type);
+
+        return $this->addRouteCollection($subCollection, $prefix);
+    }
+
+    /**
+     * Adds a route and returns it for future modification.
+     *
+     * @param string      $path       The route path
+     * @param string      $controller The route controller string
+     * @param string|null $name       The name to give this route
+     *
+     * @return Route
+     */
+    public function add($path, $controller, $name = null)
+    {
+        $route = new Route($path);
+        $route->setDefault('_controller', $controller);
+        $this->addRoute($route, $name);
+
+        return $route;
+    }
+
+    /**
+     * Returns a RouteCollectionBuilder that can be configured and then added with addBuilder().
+     *
+     * @param string $prefix A prefix to apply to all routes added to this collection
+     *
+     * @return RouteCollectionBuilder
+     */
+    public function createBuilder($prefix = null)
+    {
+        $builder = new self($this->loader);
+        $builder->setPrefix($prefix);
+
+        return $builder;
+    }
+
+    /**
+     * Add a RouteCollectionBuilder.
+     *
+     * @param RouteCollectionBuilder $builder
+     */
+    public function addBuilder(RouteCollectionBuilder $builder)
+    {
+        $this->routes[] = $builder;
+    }
+
+    /**
+     * Adds a RouteCollection directly and returns those routes in a RouteCollectionBuilder.
+     *
+     * @param RouteCollection $collection
+     * @param string|null     $prefix
+     *
+     * @return $this
+     */
+    public function addRouteCollection(RouteCollection $collection, $prefix = null)
+    {
+        // create a builder from the RouteCollection
+        $builder = $this->createBuilder($prefix);
+        foreach ($collection->all() as $name => $route) {
+            $builder->addRoute($route, $name);
+        }
+
+        foreach ($collection->getResources() as $resource) {
+            $builder->addResource($resource);
+        }
+
+        $this->addBuilder($builder);
+
+        return $builder;
+    }
+
+    /**
+     * Adds a Route object to the builder.
+     *
+     * @param Route       $route
+     * @param string|null $name
+     *
+     * @return $this
+     */
+    public function addRoute(Route $route, $name = null)
```

Comment:
In Symfony, we try to use simple `add/get/set/has` for the main object being manipulated by a collection. So, I would expect `add()` here.